source/templates/secrets.slim

```diff
@@ -6,17 +6,32 @@
 		article
 			ul.env-vars[order-by-drag models="envVarsOfSource.envVars"]
 				li[ng-repeat="envVar in envVarsOfSource.envVars"]
-					form[name="envVarForm"]
+					p.progress-indicator[progress-model="envVarsCtrl.envVarSettings[envVar.key()].progress"]
+					form[name="envVarForm" ng-if="!envVarsCtrl.envVarSettings[envVar.key()].progress.isInProgress" autocomplete="off" autocapitalize="none"]
 						.key-value-delete
-							input.key[type="text" name="key" ng-model="envVar.key" ng-model-options="{getterSetter: true}" custom-validators="{'key': Variable.isValidKey(envVar.key(), envVarsOfSource.envVars)}" placeholder="#{ data[:strings][:env_vars][:key_placeholder] }" skip-drag-order]
-							textarea.value[name="value" msd-elastic ng-model="envVar.value" ng-model-options="{getterSetter: true}" ng-trim="false" placeholder="#{ data[:strings][:env_vars][:value_placeholder] }" skip-drag-order]
-							button.delete-env-var.rebo.danger[type="button" ng-click="envVarsCtrl.removeEnvVarFromEnvVars(envVar, envVarsOfSource.envVars)"] = svg("cross")
+							input.key[type="text" name="key" ng-model="envVar.key" ng-model-options="{getterSetter: true}" ng-disabled="!envVar.isKeyChangeable" custom-validators="{'key': Variable.isValidKey(envVar.key(), envVarsOfSource.envVars)}" placeholder="#{ data[:strings][:env_vars][:key_placeholder] }" autocomplete="off" autocorrect="off" autocapitalize="none" spellcheck="false" skip-drag-order]
+							.env-var-value.secret
+								textarea.value[name="value" msd-elastic ng-model="envVarsCtrl.valueGetterSetterForEnvVar(envVar)" ng-class="{'protected' : envVar.isProtected()}" ng-disabled="!envVar.shouldShowValue" ng-model-options="{getterSetter: true}" ng-trim="false" placeholder="#{ data[:strings][:env_vars][:value_placeholder] }" skip-drag-order]
+								.lock[ng-if="requestService.mode == 'website' && !envVar.isKeyChangeable && envVar.isProtected()"] = svg("main/menuicon-secrets")
+								eye-checkbox[ng-if="requestService.mode == 'website' && !envVar.isKeyChangeable && !envVar.isProtected()" ng-model="envVar.shouldShowValue" ng-change="envVarsCtrl.shouldShowSecretEnvVarValue(envVar)" checkbox-id-expression="'secret-env-var-show-value-checkbox-' + $index"]
```

Comment:
Either:

1 - remove `ng-change` and do a getter-setter for ng-model, which in getter mode returns the shouldShowValue, in setter mode sets it to the opposite and if it is set to true, also sends the AJAX request (show an error popup if it fails).
2 - do a conditional `ng-change`: if `envVar.shouldShowValue`, execute `envVarsCtrl.showValueOfSecret(envVar)`, else null.